### PR TITLE
feat(auto-release): auto release based on conventional commit messages

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction
Introduce version and documentation auto release upon a PR merge which adhere to conventional commit.

## Resolution
* Introduced `release-please-action@v3`.
* Updated package.json `name` and `description`.

## Miscellaneous
* Dependency updates
* Typo fixes

### CODE REVIEW

File: portal-api/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "dtfs-portal-api",
```
Comment (AlexBramhill) [inline portal-api/package.json:2]: This is the only one with `dtfs` in the name as a suffix
```suggestion
  "name": "portal-api",
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          package-name: uk-export-finance/dtfs
+          default-branch: main
+          labels: ['chore', 'docs']
+          changelog-types: '[
+            { "type": "feat", "section": "Features", "hidden": false},
+            { "type": "fix", "section": "Bug Fixes", "hidden": false},
+            { "type": "chore", "section": "Miscellaneous", "hidden": false},
+            { "type": "docs", "section": "Documentation", "hidden": false}
+            { "type": "refactor", "section": "Code improvement", "hidden": false}
+            { "type": "dependabot", "section": "Dependency update", "hidden": false}
```
Comment (AlexBramhill) [inline .github/workflows/publish.yml:32]: Same comment about plural, also we use title case on `Bug Fixes` 
```suggestion
            { "type": "dependabot", "section": "Dependency Updates", "hidden": false}
```

File: external-api/package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "external-apis",
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply external-api/package.json:2]: Have updated.

File: portal-api/package.json (hunk 1 above)
Comment (abhi-markan) [reply portal-api/package.json:2]: Have updated.

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          release-type: node
+          package-name: uk-export-finance/dtfs
+          default-branch: main
+          labels: ['chore', 'docs']
+          changelog-types: '[
+            { "type": "feat", "section": "Features", "hidden": false},
+            { "type": "fix", "section": "Bug Fixes", "hidden": false},
+            { "type": "chore", "section": "Miscellaneous", "hidden": false},
+            { "type": "docs", "section": "Documentation", "hidden": false}
+            { "type": "refactor", "section": "Code improvement", "hidden": false}
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply .github/workflows/publish.yml]: Have updated.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (abhi-markan) [reply .github/workflows/publish.yml]: Have updated.

**Review by AlexBramhill — APPROVED** [review]

**Review by lmarrai-sw — APPROVED** [review]

**Pull request merged**